test/routines/levelx/xminmax.hpp

```diff
@@ -0,0 +1,168 @@
+      return status;
+    }
+    clWaitForEvents(1, &event);
+    clReleaseEvent(event);
+
+    status = Min<T>(args.n, buffers.second_scalar_uint(), args.imin_offset, buffers.x_vec(), args.x_offset, args.x_inc,
+                    &queue_plain, &event);
+    if (status != StatusCode::kSuccess) {
+      return status;
+    }
+    clWaitForEvents(1, &event);
+    clReleaseEvent(event);
+    return status;
+  }
+
+// Note: No CBlas routine exists to compare against so intead uses the RunReference1 method which works for both CBlas
+// and CLBlas
+#ifdef CLBLAST_REF_CBLAS
+  static StatusCode RunReference2(const Arguments<T>& args, BuffersHost<T>& buffers_host, Queue& queue) {
+    auto context = queue.GetContext();
+    Buffers<T> buffers{Buffer<T>{queue.GetContext(), buffers_host.x_vec.size()},
+                       CreateInvalidBuffer<T>(context, 0),
+                       CreateInvalidBuffer<T>(context, 0),
+                       CreateInvalidBuffer<T>(context, 0),
+                       CreateInvalidBuffer<T>(context, 0),
+                       CreateInvalidBuffer<T>(context, 0),
+                       CreateInvalidBuffer<T>(context, 0),
+                       Buffer<unsigned int>{queue.GetContext(), buffers_host.scalar_uint.size()},
+                       Buffer<unsigned int>{queue.GetContext(), buffers_host.second_scalar_uint.size()}};
+    HostToDevice(args, buffers, buffers_host, queue, {kBufVecX, kBufScalarUint, kBufSecondScalarUint});
+    auto status = RunReference1(args, buffers, queue);
+    DeviceToHost(args, buffers, buffers_host, queue, {kBufVecX, kBufScalarUint, kBufSecondScalarUint});
+    return status;
+  }
+#endif
+
+// Describes how to run the cuBLAS routine (for correctness/performance comparison)
+#ifdef CLBLAST_REF_CUBLAS
+  static StatusCode RunReference3(const Arguments<T>& args, BuffersCUDA<T>& buffers, Queue&) {
+    return StatusCode::kUnknownError;
+  }
+#endif
+
+  // Describes how to download the results of the computation (more importantly: which buffer)
+  static std::vector<T> DownloadResult(const Arguments<T>& args, Buffers<T>& buffers, Queue& queue) {
+    std::vector<unsigned int> result_uint(args.scalar_size, 0);
+    std::vector<unsigned int> second_result_uint(args.second_scalar_size, 0);
+    buffers.scalar_uint.Read(queue, args.scalar_size, result_uint);
+    buffers.second_scalar_uint.Read(queue, args.second_scalar_size, second_result_uint);
+    // The result is an integer. However, since the test infrastructure assumes results of
+    // type 'T' (float/double/float2/double2/half), we store the results into T instead.
+    // The values might then become meaningless, but a comparison for testing should still
+    // be valid to verify correctness.
+    auto result_as_T = static_cast<T>(result_uint[0]);
+    std::vector<T> result(args.scalar_size + args.second_scalar_size);
+    result[0] = result_as_T;
+    result[1] = static_cast<T>(second_result_uint[0]);
```

Comment:
Perhaps easier to read?
```suggestion
    std::vector<T> result(args.scalar_size + args.second_scalar_size);
    result[0] = static_cast<T>(result_uint[0]);;
    result[1] = static_cast<T>(second_result_uint[0]);
```